webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala

```diff
@@ -752,14 +754,16 @@ class NodeApiInheritedProperties(
 ) {
 
   /*
-   * Full list of node properties, including inherited ones for a node
+   * Full list of node properties, including inherited ones for a node but
+   * without hidden property.
    */
   def getNodePropertiesTree(nodeId: NodeId, renderInHtml: RenderInheritedProperties)(implicit
       qc: QueryContext
   ): IOResult[JRNodeInheritedProperties] = {
     for {
       properties       <- propRepo.getNodeProps(nodeId).notOptional(s"Node with ID '${nodeId.value}' was not found.'")
       propertiesDetails = properties
+                            .filter(_.prop.visibility == Displayed)
```

Comment:
Why not filter at the level of the target JSON class i.e. in `JRNodeInheritedProperties.fromNode(...)` ?

Reply:
filter are better done early, it makes less work for the following steps.